### PR TITLE
Improves Anomaly Logging

### PR DESCRIPTION
**Pull request by Burzah**
## What Does This PR Do
This PR improved anomaly logging, by updating messages sent to admins and the game log. It logs anomalies when they are initialized, and when the flux core anomaly detonates. It updates the message for the flux anomaly detonations. These changes do not affect anomalies spawned by SM events or Vetus. In addition, this also covers grav, cryo, and pyro anomaly detonations.

In addition, this PR also adds Blackbox logging for anomaly defusion, including type and time.  

## Why It's Good For The Game
Improves logging for administrators. 

## Images of changes
dreamseeker_MqyUsQ1374
sublime_text_CDcGfcSqXQ
sublime_text_8EsydmIwIW
q8J8jsjemb

## Testing
Game Log/Admin Message Testing:
Spawn in anomalies and observed logging.
Delamed the SM with a high EER and checked for logging changes
Fought Vetus and checked for logging changes

Blackbox Log Testing:
Spawn in an anomaly and defused.
Checked Blackbox controller information.

## Declaration
- [x] I confirm that I either do not require [pre-approval](/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.

## Changelog
NPFC

### CODE REVIEW

File: code/game/objects/effects/anomalies.dm [hunk 1]
```diff
@@ -221,12 +225,12 @@
 
 /obj/effect/anomaly/flux/detonate()
 	if(explosive)
-		explosion(src, 1, 4, 16, 18) //Low devastation, but hits a lot of stuff.
+		explosion(src, 1, 4, 16, 18, FALSE) // Set adminlog to FALSE for custom logging
+		if(drops_core)
+			log_and_message_admins("A [name] has detonated at [impact_area][ADMIN_COORDJMP(impact_area)]")
```
Comment (DGamerL) [inline code/game/objects/effects/anomalies.dm:230]: Could consider adding this to grav, pyro and cryo anomalies aswell

Comment (Burzah) [reply code/game/objects/effects/anomalies.dm:230]: Yeah I can look into updating it, I just wanted to override that explosion log for something a bit more informative to the flux anomaly itself.

Comment (Burzah) [reply code/game/objects/effects/anomalies.dm]: Updated!

File: code/game/objects/effects/anomalies.dm [hunk 2]
```diff
@@ -168,8 +170,8 @@
 	if(T && length(GLOB.gravity_generators["[T.z]"]))
 		var/obj/machinery/gravity_generator/main/G = pick(GLOB.gravity_generators["[T.z]"])
 		G.set_broken() //Requires engineering to fix the gravity generator, as it gets overloaded by the anomaly.
-
-/////////////////////
+	if(drops_core)
+		log_and_message_admins("A [name] has detonated a gravity generator")
```
Comment (DGamerL) [inline code/game/objects/effects/anomalies.dm:174]:
```suggestion
		log_and_message_admins("A [name] has detonated a gravity generator")
```

File: code/game/objects/effects/anomalies.dm [hunk 3]
```diff
@@ -221,12 +223,13 @@
 
 /obj/effect/anomaly/flux/detonate()
 	if(explosive)
-		explosion(src, 1, 4, 16, 18) //Low devastation, but hits a lot of stuff.
+		explosion(src, 1, 4, 16, 18, FALSE) // Set adminlog to FALSE for custom logging
+		if(drops_core)
+			message_admins("A [name] has detonated at [impact_area][ADMIN_COORDJMP(impact_area)]")
+			log_admin("A [name] has detonated at [impact_area]")
```
Comment (DGamerL) [inline code/game/objects/effects/anomalies.dm:229]:
```suggestion
		message_admins("A [name] has detonated at [impact_area][ADMIN_COORDJMP(impact_area)]")
		log_admin("A [name] has detonated at [impact_area]")
```
Explosive flux anomalies should always be logged
If they don't drop a core they aren't explosive (I hope)

File: code/game/objects/effects/anomalies.dm [hunk 4]
```diff
@@ -454,8 +455,9 @@
 		air.set_sleeping_agent(1000)
 		air.set_carbon_dioxide(1000)
 		T.blind_release_air(air)
-
-/////////////////////
+	if(drops_core)
```
Comment (DGamerL) [inline code/game/objects/effects/anomalies.dm:458]:
```suggestion
```

File: code/game/objects/effects/anomalies.dm [hunk 5]
```diff
@@ -221,12 +222,12 @@
 
 /obj/effect/anomaly/flux/detonate()
 	if(explosive)
-		explosion(src, 1, 4, 16, 18) //Low devastation, but hits a lot of stuff.
+		explosion(src, 1, 4, 16, 18, FALSE) // Set adminlog to FALSE for custom logging
+		message_admins("A [name] has detonated at [impact_area][ADMIN_COORDJMP(impact_area)]")
+		log_admin("A [name] has detonated at [impact_area]")
```
Comment (lewcc) [inline code/game/objects/effects/anomalies.dm:227]:
```suggestion
		message_admins("\A [name] has detonated at [impact_area][ADMIN_COORDJMP(impact_area)]")
		log_admin("\A [name] has detonated at [impact_area]")
```

File: code/game/objects/effects/anomalies.dm [hunk 6]
```diff
@@ -364,6 +362,9 @@
 /obj/effect/anomaly/pyro/detonate()
 	if(produces_slime)
 		INVOKE_ASYNC(src, PROC_REF(makepyroslime))
+	if(drops_core)
+		message_admins("A [name] has detonated at [impact_area][ADMIN_COORDJMP(impact_area)]")
+		log_admin("A [name] has detonated at [impact_area]")
```
Comment (lewcc) [inline code/game/objects/effects/anomalies.dm:367]:
```suggestion
		message_admins("\A [name] has detonated at [impact_area][ADMIN_COORDJMP(impact_area)]")
		log_admin("\A [name] has detonated at [impact_area]")
```

File: code/game/objects/effects/anomalies.dm [hunk 7]
```diff
@@ -454,8 +453,8 @@
 		air.set_sleeping_agent(1000)
 		air.set_carbon_dioxide(1000)
 		T.blind_release_air(air)
-
-/////////////////////
+		message_admins("A [name] has detonated at [impact_area][ADMIN_COORDJMP(impact_area)]")
+		log_admin("A [name] has detonated at [impact_area]")
```
Comment (lewcc) [inline code/game/objects/effects/anomalies.dm:457]:
```suggestion
		message_admins("\A [name] has detonated at [impact_area][ADMIN_COORDJMP(impact_area)]")
		log_admin("\A [name] has detonated at [impact_area]")
```

**Review by lewcc — COMMENTED** [review]
Total nitpick. Looks great otherwise!

**Comment by Burzah** [comment]
Converted back to draft, adding in blackbox logging.

**Comment by Burzah** [comment]
Update: Currently waiting for the merging of #26645

**Review by lewcc — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]

**Pull request merged**